lib/models/application.coffee

```diff
@@ -301,13 +338,20 @@ getApplicationModel = (deps, opts) ->
 	# });
 	#
 	# @example
+	# resin.models.application.getApiKey(123).then(function(apiKey) {
+	# 	console.log(apiKey);
+	# });
+	#
+	# @example
 	# resin.models.application.getApiKey('MyApp', function(error, apiKey) {
 	# 	if (error) throw error;
 	# 	console.log(apiKey);
 	# });
 	###
-	exports.getApiKey = (name, callback) ->
-		exports.get(name).then (application) ->
+	exports.getApiKey = (nameOrId, callback) ->
+		# Do a full get, not just getId, because the actual api endpoint doesn't fail if the id
+		# doesn't exist. TODO: Can use getId once https://github.com/resin-io/resin-api/issues/110 is resolved
+		exports.get(nameOrId).then (application) ->
```

Comment:
I've renamed the method, and added a deprecated `getApiKey` which we can kill off later.

I'm going to leave tweaks like `$select: id` out for now. There are (and were before this PR) quite a few places here where you could potentially make optimisations like that to reduce the data transfer, but I don't think they're necessary immediately, and they'll be easier to add once #262 is supported - I'll do this along with that. Tracked as #267.